### ISSUE TITLE
Does MetadataService really need the external Activate filter chain? And there is no filter option to disable it.

### ISSUE THREAD

**Issue by plusmancn** [apache/dubbo#8605]
- [x] I have searched the [issues](https://github.com/apache/dubbo/issues) of this repository and believe that this is not a duplicate.
- [x] I have checked the [FAQ](https://github.com/apache/dubbo/blob/master/FAQ.md) of this repository and believe that this is not a duplicate.

### Environment

* Dubbo version: the 3.0 branch of source code
* Operating System version: Mac M1
* Java version: 1.8

#### 1. It's just fine
Do nothing.

#### 2. Disable external `@Activate` filter for `org.apache.dubbo.metadata.MetadataService`
In the solution, we consider `MetadataService` as Internal serivcie, so we forbid user intercepting it.   
We can archive it in `ProtocolFilterWrapper#export` and  `ProtocolFilterWrapper#export` methods like `isRegistry` did.

```java
// => org.apache.dubbo.rpc.cluster.filter.ProtocolFilterWrapper

@Override
public <T> Exporter<T> export(Invoker<T> invoker) throws RpcException {
    if (UrlUtils.isRegistry(invoker.getUrl())) {
        return protocol.export(invoker);
    }
    return protocol.export(builder.buildInvokerChain(invoker, SERVICE_FILTER_KEY, CommonConstants.PROVIDER));
}

@Override
public <T> Invoker<T> refer(Class<T> type, URL url) throws RpcException {
    if (UrlUtils.isRegistry(url)) {
        return protocol.refer(type, url);
    }
    return builder.buildInvokerChain(protocol.refer(type, url), REFERENCE_FILTER_KEY, CommonConstants.CONSUMER);
}
```

#### 3. Add filter option for MetadataService as provider and consumer do.
If  MetadataService really needs the external Activate filter chain, but I'm doubtful about it.

**Comment by EarthChen** [comment]
I think solution 2 is better and could even be considered entirely as an Internal service

**Comment by plusmancn** [comment]
I will submit a PR after Dubbo 3.0.3 release.

**Comment by EarthChen** [comment]
> I will submit a PR after Dubbo 3.0.3 release.

We're doing something similar in the tri protocol. Add some internal services, such as health checks, reflections, etc., and now it's the practice to use proxy to generate invoker, put it in the service repository, and put it in the pathresolve

**Comment by plusmancn** [comment]
Yep,  we need something like a whitelist or annotation to identify which services are internal services.

**Comment by EarthChen** [comment]
> Yep, we need something like a whitelist or annotation to identify which services are internal services.
> 
> But I'm not sure whether the multi-instances feature will break some conventions, so just wait for the 3.0.3 release.

I think we can consider implementing an internal service creation process instead of creating from serviceconfig. This is because I do not have this class at the rpc layer.

**Comment by plusmancn** [comment]
> implementing an internal service creation process instead of creating from serviceconfig

What's the benefit against the cost? The original purpose was only to guarantee the safety of internal service, and moreover `ProtocolFilterWrapper` have already implemented some feature which is similar to whitelist.

```java
// => org.apache.dubbo.rpc.cluster.filter.ProtocolFilterWrapper

@Override
public <T> Exporter<T> export(Invoker<T> invoker) throws RpcException {
   // => Here, DON'T wrap with filter chain
    if (UrlUtils.isRegistry(invoker.getUrl())) {
        return protocol.export(invoker);
    }
    return protocol.export(builder.buildInvokerChain(invoker, SERVICE_FILTER_KEY, CommonConstants.PROVIDER));
}

@Override
public <T> Invoker<T> refer(Class<T> type, URL url) throws RpcException {
    if (UrlUtils.isRegistry(url)) {
        return protocol.refer(type, url);
    }
    return builder.buildInvokerChain(protocol.refer(type, url), REFERENCE_FILTER_KEY, CommonConstants.CONSUMER);
}

```

**Comment by EarthChen** [comment]
This is another problem. In fact, I put forward this idea because of the consideration of the internal services of the current protocol, there are no things in the serviceconfig of the RPC layer, and the implementation is not elegant. Metaservice is currently initialized in the bootstrap. In fact, these problems will not be encountered, but all internal services cannot be initialized in the bootstrap, right?

**Comment by EarthChen** [comment]
> > implementing an internal service creation process instead of creating from serviceconfig
> 
> What's the benefit against the cost? The original purpose was only to guarantee the safety of internal service, and moreover `ProtocolFilterWrapper` have already implemented some feature which is similar to whitelist.
> 
> ```java
> // => org.apache.dubbo.rpc.cluster.filter.ProtocolFilterWrapper
> 
> @Override
> public <T> Exporter<T> export(Invoker<T> invoker) throws RpcException {
>    // => Here, DON'T wrap with filter chain
>     if (UrlUtils.isRegistry(invoker.getUrl())) {
>         return protocol.export(invoker);
>     }
>     return protocol.export(builder.buildInvokerChain(invoker, SERVICE_FILTER_KEY, CommonConstants.PROVIDER));
> }
> 
> @Override
> public <T> Invoker<T> refer(Class<T> type, URL url) throws RpcException {
>     if (UrlUtils.isRegistry(url)) {
>         return protocol.refer(type, url);
>     }
>     return builder.buildInvokerChain(protocol.refer(type, url), REFERENCE_FILTER_KEY, CommonConstants.CONSUMER);
> }
> ```

You're right about this. I agree to do so

**Comment by plusmancn** [comment]
@EarthChen 

> In fact, these problems will not be encountered

It's common that users add `@Activate` to their self-defined filter, such as sentinel.

> all internal services cannot be initialized in the bootstrap

As far as I know, internal services like `Metaservice` are just normal services that can be described by `ServiceConfig`.
If you are sticking to separate it from `ServiceConfig`, maybe  @AlbumenJ @chickenlj  can answer your question and make a decision.

**Comment by plusmancn** [comment]
Related to #8666
